As per https://github.com/conda-forge/glib-feedstock/issues/40, the new Meson
build system can cause problems on Linux. When installing executables
(including shared libraries), Meson edits their RPATHs. When no
"install_rpath" setting has been configured, Meson removes the RPATH entry.
First, this can break our executables. Second, historically, the way that
Meson does so was legal, but ends up altering the file structure in a way
that a few naive programs can't handle. One of those programs is ldconfig, and
its confusion can result in it trying to create files with junk names, which
then confuse downstream tools.

So we the install rpath to something nonempty.

diff --git a/glib/meson.build b/glib/meson.build
index 6fc56da..4cfc08d 100644
--- a/glib/meson.build
+++ b/glib/meson.build
@@ -257,6 +257,7 @@ libglib = library('glib-2.0',
   version : library_version,
   soversion : soversion,
   install : true,
+  install_rpath : glib_libdir,
   # intl.lib is not compatible with SAFESEH
   link_args : [noseh_link_args, glib_link_flags, win32_ldflags],
   include_directories : configinc,
@@ -315,6 +316,7 @@ if host_system == 'windows'
 else
   gtester = executable('gtester', 'gtester.c',
     install : true,
+    install_rpath : glib_libdir,
     include_directories : configinc,
     dependencies : [libglib_dep])
 endif
diff --git a/gio/meson.build b/gio/meson.build
index 7f2c08e..871075b 100644
--- a/gio/meson.build
+++ b/gio/meson.build
@@ -423,6 +423,7 @@ if host_system != 'windows'
 
     executable('gio-launch-desktop', 'gio-launch-desktop.c',
       install : true,
+      install_rpath : glib_libdir,
       c_args : gio_c_args,
       # intl.lib is not compatible with SAFESEH
       link_args : noseh_link_args)
@@ -798,6 +799,7 @@ libgio = library('gio-2.0',
   version : library_version,
   soversion : soversion,
   install : true,
+  install_rpath : glib_libdir,
   include_directories : [configinc, gioinc],
   #  '$(gio_win32_res_ldflag)',
   dependencies : [libz_dep, libdl_dep, libmount_dep, libglib_dep,
@@ -919,6 +921,7 @@ gio_tool_sources = [
 
 executable('gio', gio_tool_sources,
   install : true,
+  install_rpath : glib_libdir,
   c_args : gio_c_args,
   # intl.lib is not compatible with SAFESEH
   link_args : noseh_link_args,
@@ -926,12 +929,14 @@ executable('gio', gio_tool_sources,
 
 executable('gresource', 'gresource-tool.c',
   install : true,
+  install_rpath : glib_libdir,
   # intl.lib is not compatible with SAFESEH
   link_args : noseh_link_args,
   dependencies : [libelf, libgio_dep, libgobject_dep, libgmodule_dep, libglib_dep])
 
 gio_querymodules = executable('gio-querymodules', 'gio-querymodules.c', 'giomodule-priv.c',
   install : true,
+  install_rpath : glib_libdir,
   c_args : gio_c_args,
   # intl.lib is not compatible with SAFESEH
   link_args : noseh_link_args,
@@ -940,6 +945,7 @@ gio_querymodules = executable('gio-querymodules', 'gio-querymodules.c', 'giomodu
 glib_compile_schemas = executable('glib-compile-schemas',
   [gconstructor_as_data_h, 'gvdb/gvdb-builder.c', 'glib-compile-schemas.c'],
   install : true,
+  install_rpath : glib_libdir,
   # intl.lib is not compatible with SAFESEH
   link_args : noseh_link_args,
   dependencies : [libgio_dep, libgobject_dep, libgmodule_dep, libglib_dep])
@@ -947,6 +953,7 @@ glib_compile_schemas = executable('glib-compile-schemas',
 glib_compile_resources = executable('glib-compile-resources',
   [gconstructor_as_data_h, 'gvdb/gvdb-builder.c', 'glib-compile-resources.c'],
   install : true,
+  install_rpath : glib_libdir,
   c_args : gio_c_args,
   # intl.lib is not compatible with SAFESEH
   link_args : noseh_link_args,
@@ -954,6 +961,7 @@ glib_compile_resources = executable('glib-compile-resources',
 
 executable('gsettings', 'gsettings-tool.c',
   install : true,
+  install_rpath : glib_libdir,
   c_args : gio_c_args,
   # intl.lib is not compatible with SAFESEH
   link_args : noseh_link_args,
@@ -966,6 +974,7 @@ install_data(['gschema.loc', 'gschema.its'],
 
 executable('gdbus', 'gdbus-tool.c',
   install : true,
+  install_rpath : glib_libdir,
   c_args : gio_c_args,
   # intl.lib is not compatible with SAFESEH
   link_args : noseh_link_args,
@@ -974,6 +983,7 @@ executable('gdbus', 'gdbus-tool.c',
 if host_system != 'windows' and not glib_have_cocoa
   executable('gapplication', 'gapplication-tool.c',
     install : true,
+    install_rpath : glib_libdir,
     c_args : gio_c_args,
     # intl.lib is not compatible with SAFESEH
     link_args : noseh_link_args,
diff --git a/gobject/meson.build b/gobject/meson.build
index d8d421d..1a7d320 100644
--- a/gobject/meson.build
+++ b/gobject/meson.build
@@ -67,6 +67,7 @@ libgobject = library('gobject-2.0',
   version : library_version,
   soversion : soversion,
   install : true,
+  install_rpath : glib_libdir,
   include_directories : [configinc],
   dependencies : [libffi_dep, libglib_dep],
   c_args : ['-DG_LOG_DOMAIN="GLib-GObject"', '-DGOBJECT_COMPILATION'] + glib_hidden_visibility_args,
@@ -111,6 +112,7 @@ endforeach
 
 executable('gobject-query', 'gobject-query.c',
   install : true,
+  install_rpath : glib_libdir,
   dependencies : [libglib_dep, libgobject_dep])
 
 install_data('gobject_gdb.py', install_dir : join_paths(glib_pkgdatadir, 'gdb'))
diff --git a/gmodule/meson.build b/gmodule/meson.build
index 8bb6189..be254f3 100644
--- a/gmodule/meson.build
+++ b/gmodule/meson.build
@@ -100,6 +100,7 @@ libgmodule = library('gmodule-2.0',
   version : library_version,
   soversion : soversion,
   install : true,
+  install_rpath : glib_libdir,
   include_directories : [configinc, gmoduleinc],
   dependencies : [libdl_dep, libglib_dep],
   c_args : ['-DG_LOG_DOMAIN="GModule"', '-DG_DISABLE_DEPRECATED'] + glib_hidden_visibility_args,
diff --git a/gthread/meson.build b/gthread/meson.build
index 515479b..0761b04 100644
--- a/gthread/meson.build
+++ b/gthread/meson.build
@@ -17,6 +17,7 @@ libgthread = library('gthread-2.0',
   version : library_version,
   soversion : soversion,
   install : true,
+  install_rpath : glib_libdir,
   dependencies : [libglib_dep],
   c_args : ['-DG_LOG_DOMAIN="GThread"' ] + glib_hidden_visibility_args,
   link_args : glib_link_flags,
